### PR TITLE
make spotlight search safer for bad inputs

### PR DESCRIPTION
**Pull request by derrickstaten**
@Sarkar @aaustin

### CODE REVIEW

File: Branch-SDK/Branch-SDK/BNCContentDiscoveryManager.m [hunk 1]
```diff
@@ -135,14 +135,17 @@ - (void)indexContentWithTitle:(NSString *)title description:(NSString *)descript
         }
         return;
     }
-    if (!title) {
+    if (!title || ![title isKindOfClass:[NSString class]]) {
```
Comment (shortstuffsushi) [inline Branch-SDK/Branch-SDK/BNCContentDiscoveryManager.m]: The method def here specifically calls for an `NSString`, type checking that seems unnecessary.

**Comment by shortstuffsushi** [comment]
This whole thing looks excessive. What are the use cases for people passing in specifically bad typed values into a method with explicit types?

**Comment by aaustin** [comment]
:+1:

**Comment by derrickstaten** [comment]
I agree that this is excessive. I don't see any real value-add here. I'm going to close this for now.

**Pull request closed without merging**